javascript/test/basic-tests.js

```diff
@@ -8,103 +9,137 @@ const { stdin } = require('mock-stdin');
+    describe('Runs with non-null exit codes. Tag: Assignment1, Assignment2, Assignment3, Assignment4, Assignment5', function () {
+
+        var tempDir;
+        var minesweeperProcess;
 
-        beforeEach("Ensure no simple.cfg before execution.", function(){
-            if( fs.existsSync("simple.cfg")){
-                fs.unlinkSync("simple.cfg");
-            }
+        beforeEach("Create temporary folder for file creation", function () {
+            tempDir = fs.mkdtempSync(os.tmpdir() + path.sep);
         });
 
-        afterEach("Ensure no simple.cfg after execution", function(){
-            if( fs.existsSync("simple.cfg")){
-                fs.unlinkSync("simple.cfg");
-            }
+        afterEach("Remove temporary folder for file creation", function () {
+            // storing the global variable into a local one for each test might be a bit of a hack,
+            // but because of the asynchronous behavior there is not really another chance (to the best of my knowledge)
+            // unless abstaining from the usage of before/afterEach.
+            var tempDirLocal = tempDir;
+            minesweeperProcess.on('close', () => {
+                deleteFolderWithFiles(tempDirLocal);
+            });
         });
 
-        it('Minesweeper should exit with code 1 if input file is not provided', function() {
-            var exitCode = minesweeper.main()
-            assert.strictEqual(exitCode, 1, "Wrong exit code for missing board configuration file")
+        it('Minesweeper should exit with code 1 if input file is not provided', function () {
+            minesweeperProcess = spawn('node', [minesweeperHome + "minesweeper.js"]);
+
+            minesweeperProcess.on('close', (exitCode) => {
+                assert.strictEqual(exitCode, 1, "Wrong exit code for missing board configuration file")
+            });
         });
 
-        it('Minesweeper should exit with code 1 if input file does not exist', function() {
-            var exitCode = minesweeper.main("simple.cfg")
-            assert.strictEqual(exitCode, 1, "Wrong exit code for not existent board configuration file")
+        it('Minesweeper should exit with code 1 if input file does not exist', function () {
+            minesweeperProcess = spawn('node', [minesweeperHome + "minesweeper.js", "simple.cfg"]);
+
+            minesweeperProcess.on('close', (exitCode) => {
+                assert.strictEqual(exitCode, 1, "Wrong exit code for not existent board configuration file")
+            });
         });
-        
-        it('Minesweeper should exit with code 2 if input file is invalid (empty)', function() {
+
+        it('Minesweeper should exit with code 2 if input file is invalid (empty)', function () {
             // Make sure there's an empty "simple.cfg" file (see https://flaviocopes.com/how-to-create-empty-file-node/)
-            fs.closeSync(fs.openSync("simple.cfg", 'w'))
+            fs.closeSync(fs.openSync(tempDir + "/simple.cfg", 'w'))
```

Comment:
See comment above on defining `tempDir` as `os.tmpdir() + path.sep`. I suspec that the `/` might be superfluous/wrong?